src/Features/CSharpTest/GenerateMethod/GenerateMethodTests.cs

```diff
@@ -10839,4 +10839,255 @@ private static void Goo()
+                }
+            }
+            """,
+            """
+            using System;
+            
+            class C
+            {
+                int M()
+                {
+                    var (x, y, (z, w)) = NewExpr();
+                }
+            
+                private (object x, object y, (object z, object w)) NewExpr()
+                {
+                    throw new NotImplementedException();
+                }
+            }
+            """);
+    }
+
+    [Fact, WorkItem("https://github.com/dotnet/roslyn/issues/71980")]
+    public async Task AssignToTuple5()
+    {
+        await TestInRegularAndScriptAsync(
+            """
+            using System;
+
+            class C
+            {
+                int M()
+                {
+                    var (x, y, (z, w, )) = [|NewExpr()|];
+                }
+            }
+            """,
+            """
+            using System;
+            
+            class C
+            {
+                int M()
+                {
+                    var (x, y, (z, w, )) = NewExpr();
+                }
+            
+                private object NewExpr()
+                {
+                    throw new NotImplementedException();
+                }
+            }
+            """);
+    }
+
+    [Fact, WorkItem("https://github.com/dotnet/roslyn/issues/71980")]
+    public async Task AssignToTuple6()
+    {
+        await TestInRegularAndScriptAsync(
+            """
+            using System;
+
+            class C
+            {
+                int M()
+                {
+                    var (x, y, (z, w), ) = [|NewExpr()|];
+                }
+            }
+            """,
+            """
+            using System;
+            
+            class C
+            {
+                int M()
+                {
+                    var (x, y, (z, w), ) = NewExpr();
+                }
+            
+                private object NewExpr()
+                {
+                    throw new NotImplementedException();
+                }
+            }
+            """);
+    }
+
+    [Fact, WorkItem("https://github.com/dotnet/roslyn/issues/71980")]
+    public async Task AssignToTuple7()
+    {
+        await TestMissingInRegularAndScriptAsync(
+            """
+            using System;
+            using System.Threading.Tasks;
+
+            class C
+            {
+                async Task<int> M()
+                {
+                    var ([|NewExpr|]) = await G();
+                }
+
+                Task<(int, string)> G() => default;
+            }
+            """);
+    }
+
+    [Fact, WorkItem("https://github.com/dotnet/roslyn/issues/71980")]
+    public async Task AssignToTuple8()
+    {
+        await TestInRegularAndScriptAsync(
+            """
+            using System;
+
+            class C
+            {
+                int M()
+                {
+                    var (x) = [|NewExpr()|];
```

Comment:
note: This is hte one test that would trigger the crash.  the rest are still good though for coverage and verification of strange cases.